### PR TITLE
fix(license): make search for licenses case insensitive 

### PR DESCRIPTION
**Pull request by Herschdorfer**
*removed case sensitive license search
*added search also in shortname

related to
#478 #348 #458

### CODE REVIEW

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/Sw360Portlet.java [hunk 1]
```diff
@@ -198,7 +200,10 @@ public void dealWithLicenseAction(ResourceRequest request, ResourceResponse resp
                         @Override
                         public boolean apply(License input) {
                             String fullname = input.getFullname();
-                            return !isNullOrEmpty(fullname) && fullname.contains(searchText);
+                            String shortname = input.getShortname();
+                            return !isNullOrEmpty(fullname)
```
Comment (alexbrdn) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/Sw360Portlet.java]: `isNullOrEmpty` check is not necessary if you're using `StringUtils.containsIgnoreCase` as it checks for null

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/Sw360Portlet.java [hunk 2]
```diff
@@ -198,7 +200,10 @@ public void dealWithLicenseAction(ResourceRequest request, ResourceResponse resp
...
+                                   && ((StringUtils.containsIgnoreCase(searchText, fullname)
```
Comment (alexbrdn) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/Sw360Portlet.java]: This searches for `fullname` in `searchText`. It was the other way around before.
I think you should flip the arguments in this and the next line.

**Comment by Herschdorfer** [comment]
your right, pushed the changes

**Comment by mcjaeger** [comment]
yes, squash ... please let us know if help would be helpful. Another thing would be to have a nice commit message ("conventional commits"), like

fix(license): make search for licenses case insensitive

more details here: https://github.com/sw360/sw360portal/wiki/Dev-DoD-and-Style

**Comment by mcjaeger** [comment]
(that all said,  the change to the search is a super useful contribution)

**Comment by Herschdorfer** [comment]
hey,
i combined it and changed the commit message. 
np, will help me also ;)

**Comment by alexbrdn** [comment]
This doesn't work with shortnames :( Bummer, I know.
The reason is that `LicenseSummary` doesn't fill in the field `shortname`. 
This patch solves it:
```diff
index 8a8079d..a8e07ad 100644
--- a/backend/src-common/src/main/java/org/eclipse/sw360/components/summary/LicenseSummary.java
+++ b/backend/src-common/src/main/java/org/eclipse/sw360/components/summary/LicenseSummary.java
@@ -34,6 +34,7 @@ public class LicenseSummary extends DocumentSummary<License> {
                 copyField(document, copy, _Fields.LICENSE_TYPE);
             default:
                 copyField(document, copy, _Fields.ID);
+                copy.setShortname(document.getId());
                 copyField(document, copy, _Fields.FULLNAME);
                 copyField(document, copy, _Fields.LICENSE_TYPE_DATABASE_ID);
         }
```
@Herschdorfer, please update the PR with this one line and we'll be able to merge it.

**Comment by Herschdorfer** [comment]
Would 
`copyField(document, copy, _Fields.SHORTNAME);`

not be the better choice?

**Comment by alexbrdn** [comment]
I don't think so. As it stands currently, a license's shortname _is_ its id (See #135). One cannot guarantee that a license from the database has a shortname at all. Therefore, if we are to use the field shortname in the code, we ought to make sure it is equal to license's id.

That said, there's an idea floating in the room that we should pull shortname and id back apart.

**Comment by Herschdorfer** [comment]
ok pitty...
i changed the commit, we need to have this in mind when changing the id/shortname stuff.

**Pull request merged**